frontend/src/libs/components/select/select.tsx

```diff
@@ -0,0 +1,110 @@
+import {
+  type Control,
+  type FieldErrors,
+  type FieldPath,
+  type FieldValues,
+} from 'react-hook-form';
+import { type GroupBase, type Props, type StylesConfig } from 'react-select';
+import { default as ReactSelect } from 'react-select';
+
+import { useCallback, useFormController } from '~/libs/hooks/hooks.js';
+import { type SelectOption } from '~/libs/types/types.js';
+
+import {
+  DropdownIndicator,
+  IndicatorSeparator,
+} from './libs/components/components.js';
+import { Colors } from './libs/enums/colors.enum.js';
+import cssStyles from './styles.module.scss';
+
+type Properties<
+  T extends FieldValues,
+  IsMulti extends boolean = false,
+  Group extends GroupBase<SelectOption> = GroupBase<SelectOption>,
+> = Props<SelectOption, IsMulti, Group> & {
+  name: FieldPath<T>;
+  control: Control<T, null>;
+  errors: FieldErrors<T>;
+  label: string;
+};
+
+const Select = <
+  T extends FieldValues,
+  IsMulti extends boolean = false,
+  Group extends GroupBase<SelectOption> = GroupBase<SelectOption>,
+>({
+  placeholder = 'Search...',
+  styles,
+  name,
+  control,
+  errors,
+  label,
+  ...restProperties
+}: Properties<T, IsMulti, Group>): JSX.Element => {
+  const { field } = useFormController({ name, control });
+  const error = errors[name]?.message;
+  const hasError = Boolean(error);
+
+  const defaultStyles: StylesConfig<SelectOption, IsMulti, Group> = {
```

Comment:
I tried to do it but there are type issues if I move it outside.  I can pass SelectOption and Group outside but not isMulti.  
I tried to skip it but it's a second variable so something like StyleConfig<SelectOption, Group> doesnt work because it expects second type variable to extend boolean (i.e to be isMulti)

Reply:
Maybe then we can wrap `defaultStyles` into `getDefaultStyles` helper - then method could have generic typing that would be identical to generic passed for Select component